File: app/src/main/java/com/handstandsam/maintainableespresso/features/home/HomeRVAdapter.java
package com.handstandsam.maintainableespresso.features.home;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.handstandsam.maintainableespresso.R;
import com.handstandsam.maintainableespresso.models.Category;

import java.util.ArrayList;
import java.util.List;

class HomeRVAdapter extends RecyclerView.Adapter<CategoryViewHolder> {
    private List<Category> categories = new ArrayList<>();

    public HomeRVAdapter() {
    }

    @Override
    public CategoryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_recyclerview_category_row, parent, false);
        return new CategoryViewHolder(view);
    }

    public void setData(List<Category> categories) {
        this.categories = categories;
        notifyDataSetChanged();
    }

    @Override
    public void onBindViewHolder(CategoryViewHolder holder, int position) {
        holder.bindData(categories.get(position), position);
    }

    @Override
    public int getItemCount() {
        return categories.size();
    }
}

File: app/src/main/java/com/handstandsam/shoppingapp/features/home/HomeRVAdapter.kt
package com.handstandsam.shoppingapp.features.home

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup

import com.handstandsam.shoppingapp.R
import com.handstandsam.shoppingapp.models.Category

import java.util.ArrayList

internal class HomeRVAdapter : RecyclerView.Adapter<CategoryViewHolder>() {
    private var categories: List<Category> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.view_recyclerview_category_row, parent, false)
        return CategoryViewHolder(view)
    }

    fun setData(categories: List<Category>) {
        this.categories = categories
        notifyDataSetChanged()
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        holder.bindData(categories[position], position)
    }

    override fun getItemCount(): Int {
        return categories.size
    }
}
